File: bootcamp_c_cpp_dsa-main/bootcamp_c_cpp_dsa-main/C+DS/codes/Subarray/p2.c
//Given an array of size N  print the sum of every single subarray


#include<stdio.h>
void main(){

	int arr[]={2,4,1,3};

	for(int i=0;i<4;i++){
		for(int j=i;j<4;j++){
			int sum=0;
			for(int k=i;k<=j;k++){
				sum=sum+arr[k];
			}
			printf("%d\n",sum);
		}
	}
}

File: java-main/java-main/AB/p2.java
/*Maximize sum(arr[i]*i) of an Array
Given an array A of N integers. Your task is to write a program to find the
maximum value of ∑arr[i]*i, where i = 0, 1, 2,., n 1.
You are allowed to rearrange the elements of the array.
Note: Since output could be large, hence module 109+7 and then print answer.
Example 1:
Input : Arr[] = {5, 3, 2, 4, 1}
Output : 40*/


import java.util.Arrays;

class MaximizeSum {
    static long maxSum(long[] arr) {
        long mod = 1000000007; 
        int n = arr.length;

        Arrays.sort(arr);

        long result = 0;

        for (int i = 0; i < n; i++) {
            result = (result + (arr[i] * i)) % mod;
        }

        return result;
    }

    public static void main(String[] args) {
        long[] arr = {5, 3, 2, 4, 1};
        System.out.println(maxSum(arr)); 					
    }
}

